dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramIndicatorService.java

```diff
@@ -504,9 +521,18 @@ public String filterIsValid( String filter )
             return expr;
         }
 
-        if ( !ExpressionUtils.isBoolean( expr, null ) )
+        // Catch any exceptions from interpreting the expression, incase the expression is invalid.
+        try
         {
-            return ProgramIndicator.FILTER_NOT_EVALUATING_TO_TRUE_OR_FALSE;
+            if ( !ExpressionUtils.isBoolean( expr, null ) )

```

Comment:
I would suggest we move the exception catching into the isBoolean method. That way we encapsulate the check rather than than spreading it throughout the code.